qiita_pet/handlers/study_handlers/listing_handlers.py

```diff
@@ -42,13 +43,40 @@ def _get_shared_links_for_study(study):
     return ", ".join(shared)
 
 
-def _build_study_info(user, results=None):
-    """builds list of dicts for studies table, with all html formatted"""
+def _build_study_info(user, study_proc=None, proc_samples=None):
+    """builds list of dicts for studies table, with all html formatted
+
+    Parameters
+    ----------
+    user : User object
+        logged in user
+    study_proc : dict of lists
+        Dictionary keyed on study_id that lists all processed data associated
+        with that study.
+    proc_samples : dict of lists
+        Dictionary keyed on proc_data_id that lists all samples associated with
+        that processed data.
+
+    Notes
+    -----
+    Both study_proc and proc_samples must be passed, or neither passed.
```

Comment:
I would suggest also adding a comment on the description of each parameter: `must be provided if 'proc_samples' is provided`; or something along those lines